Error with qat_compress.c and qat_crypt.c, initialization from incompatible pointer type during compilation with Intel QAT

<!-- Please fill out the following template, which will help other contributors address your issue. -->

<!--
Thank you for reporting an issue.

*IMPORTANT* - Please search our issue tracker *before* making a new issue.
If you cannot find a similar issue, then create a new issue.
https://github.com/zfsonlinux/zfs/issues 

*IMPORTANT* - This issue tracker is for *bugs* and *issues* only.
Please search the wiki and the mailing list archives before asking 
questions on the mailing list.
https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists

Please fill in as much of the template as possible.
-->

### System information
<!--  add version after "|" character -->
Type | Version/Name
 --- | --- 
Distribution Name	| Debian
Distribution Version	| 10
Linux Kernel	| 4.19.0-5
Architecture	| amd64
ZFS Version	| 0.8.1
SPL Version	| N/A
<!-- 
Commands to find ZFS/SPL versions:
modinfo zfs | grep -iw version
modinfo spl | grep -iw version 
-->

### Describe the problem you're observing
Can't compile ZFS with Intel QAT driver

### Describe how to reproduce the problem
Download/Install Intel QAT driver from 01.org, then try to compile ZFS with QAT:
```
# ./configure --with-qat=/usr/local/QAT
# make
```

### Include any warning/errors/backtraces from the system logs
<!-- 
*IMPORTANT* - Please mark logs and text output from terminal commands 
or else Github will not display them correctly. 
An example is provided below.

Example:
```
this is an example how log text should be marked (wrap it with ```)
```
-->

```
/usr/local/zfs-0.8.1# make
[snip]

  CC [M]  /usr/local/zfs-0.8.1/module/zfs/vdev_raidz_math_sse2.o
  CC [M]  /usr/local/zfs-0.8.1/module/zfs/vdev_raidz_math_ssse3.o
In file included from /usr/src/linux-headers-4.19.0-5-common/include/linux/module.h:18,
                 from /usr/local/zfs-0.8.1/include/linux/mod_compat.h:29,
                 from /usr/local/zfs-0.8.1/module/zfs/qat_compress.c:27:
/usr/local/zfs-0.8.1/module/zfs/qat_compress.c:571:45: error: initialization of ‘int (*)(const char *, const struct kernel_param *)’ from incompatible pointer type ‘int (*)(const char *, struct kernel_param *)’ [-Werror=incompatible-pointer-types]
 module_param_call(zfs_qat_compress_disable, param_set_qat_compress,
                                             ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux-headers-4.19.0-5-common/include/linux/moduleparam.h:233:24: note: in definition of macro ‘module_param_call’
   { .flags = 0, .set = _set, .get = _get };  \
                        ^~~~
/usr/local/zfs-0.8.1/module/zfs/qat_compress.c:571:45: note: (near initialization for ‘__param_ops_zfs_qat_compress_disable.set’)
 module_param_call(zfs_qat_compress_disable, param_set_qat_compress,
                                             ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux-headers-4.19.0-5-common/include/linux/moduleparam.h:233:24: note: in definition of macro ‘module_param_call’
   { .flags = 0, .set = _set, .get = _get };  \
                        ^~~~
  CC [M]  /usr/local/zfs-0.8.1/module/zfs/vdev_raidz_math_avx2.o
  CC [M]  /usr/local/zfs-0.8.1/module/zfs/vdev_raidz_math_avx512f.o
  CC [M]  /usr/local/zfs-0.8.1/module/zfs/vdev_raidz_math_avx512bw.o
In file included from /usr/src/linux-headers-4.19.0-5-common/include/linux/module.h:18,
                 from /usr/local/zfs-0.8.1/include/linux/mod_compat.h:29,
                 from /usr/local/zfs-0.8.1/module/zfs/qat_crypt.c:34:
/usr/local/zfs-0.8.1/module/zfs/qat_crypt.c:624:44: error: initialization of ‘int (*)(const char *, const struct kernel_param *)’ from incompatible pointer type ‘int (*)(const char *, struct kernel_param *)’ [-Werror=incompatible-pointer-types]
 module_param_call(zfs_qat_encrypt_disable, param_set_qat_encrypt,
                                            ^~~~~~~~~~~~~~~~~~~~~
/usr/src/linux-headers-4.19.0-5-common/include/linux/moduleparam.h:233:24: note: in definition of macro ‘module_param_call’
   { .flags = 0, .set = _set, .get = _get };  \
                        ^~~~
/usr/local/zfs-0.8.1/module/zfs/qat_crypt.c:624:44: note: (near initialization for ‘__param_ops_zfs_qat_encrypt_disable.set’)
 module_param_call(zfs_qat_encrypt_disable, param_set_qat_encrypt,
                                            ^~~~~~~~~~~~~~~~~~~~~
/usr/src/linux-headers-4.19.0-5-common/include/linux/moduleparam.h:233:24: note: in definition of macro ‘module_param_call’
   { .flags = 0, .set = _set, .get = _get };  \
                        ^~~~
/usr/local/zfs-0.8.1/module/zfs/qat_crypt.c:628:45: error: initialization of ‘int (*)(const char *, const struct kernel_param *)’ from incompatible pointer type ‘int (*)(const char *, struct kernel_param *)’ [-Werror=incompatible-pointer-types]
 module_param_call(zfs_qat_checksum_disable, param_set_qat_checksum,
                                             ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux-headers-4.19.0-5-common/include/linux/moduleparam.h:233:24: note: in definition of macro ‘module_param_call’
   { .flags = 0, .set = _set, .get = _get };  \
                        ^~~~
/usr/local/zfs-0.8.1/module/zfs/qat_crypt.c:628:45: note: (near initialization for ‘__param_ops_zfs_qat_checksum_disable.set’)
 module_param_call(zfs_qat_checksum_disable, param_set_qat_checksum,
                                             ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux-headers-4.19.0-5-common/include/linux/moduleparam.h:233:24: note: in definition of macro ‘module_param_call’
   { .flags = 0, .set = _set, .get = _get };  \
                        ^~~~
cc1: some warnings being treated as errors
make[7]: *** [/usr/src/linux-headers-4.19.0-5-common/scripts/Makefile.build:308: /usr/local/zfs-0.8.1/module/zfs/qat_compress.o] Error 1
make[7]: *** Waiting for unfinished jobs....
cc1: some warnings being treated as errors
make[7]: *** [/usr/src/linux-headers-4.19.0-5-common/scripts/Makefile.build:308: /usr/local/zfs-0.8.1/module/zfs/qat_crypt.o] Error 1
make[6]: *** [/usr/src/linux-headers-4.19.0-5-common/scripts/Makefile.build:549: /usr/local/zfs-0.8.1/module/zfs] Error 2
make[5]: *** [/usr/src/linux-headers-4.19.0-5-common/Makefile:1539: _module_/usr/local/zfs-0.8.1/module] Error 2
make[4]: *** [Makefile:146: sub-make] Error 2
make[3]: *** [Makefile:8: all] Error 2
make[3]: Leaving directory '/usr/src/linux-headers-4.19.0-5-amd64'
make[2]: *** [Makefile:30: modules] Error 2
make[2]: Leaving directory '/usr/local/zfs-0.8.1/module'
make[1]: *** [Makefile:841: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/zfs-0.8.1'
make: *** [Makefile:710: all] Error 2
/usr/local/zfs-0.8.1# 
```

--- a/module/zfs/qat_compress.c
+++ b/module/zfs/qat_compress.c
@@ -547,7 +547,7 @@ qat_compress(qat_compress_dir_t dir, char *src, int src_len,
 }
 
 static int
-param_set_qat_compress(const char *val, struct kernel_param *kp)
+param_set_qat_compress(const char *val, zfs_kernel_param_t *kp)
 {
 	int ret;
 	int *pvalue = kp->arg;

--- a/module/zfs/qat_crypt.c
+++ b/module/zfs/qat_crypt.c
@@ -578,7 +578,7 @@ fail:
 }
 
 static int
-param_set_qat_encrypt(const char *val, struct kernel_param *kp)
+param_set_qat_encrypt(const char *val, zfs_kernel_param_t *kp)
 {
 	int ret;
 	int *pvalue = kp->arg;
@@ -600,7 +600,7 @@ param_set_qat_encrypt(const char *val, struct kernel_param *kp)
 }
 
 static int
-param_set_qat_checksum(const char *val, struct kernel_param *kp)
+param_set_qat_checksum(const char *val, zfs_kernel_param_t *kp)
 {
 	int ret;
 	int *pvalue = kp->arg;

